staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
+}
+
+// BearerTokenSource is used to generate
+type BearerTokenSource interface {
+
+	// Token returns a bearer token to add to all requests sent by the REST client.
```

Comment:
how does this interact with requests that already have an authorization header set (c.f. https://github.com/kubernetes/kubernetes/pull/45529 and https://github.com/kubernetes/kubernetes/issues/44476)?